oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/FunctionIndexTest.java

```diff
@@ -240,6 +241,114 @@ public void upperCase() throws Exception {
+        a.setProperty("foo", "hello");
+
+        a = test.addChild("n2");
+        a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+        a.setProperty("foo", "World!");
+        a = test.addChild("n3");
+        a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+        a.setProperty("foo", "Hallo");
+        a = test.addChild("n4");
+        a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+        a.setProperty("foo", "10%");
+        a = test.addChild("n5");
+        a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+        a.setProperty("foo", "10 percent");
+
+        a = test.addChild("n0");
+        a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+        a = test.addChild("n9");
+        a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+
+        String query = "select a.[foo]\n" +
+                "\t  from [nt:unstructured] as a\n" +
+                "\t  where a.foo is not null and isdescendantnode(a , '/test') order by upper(a.foo)";
+
+        root.commit();
+
+        assertThat(explain(query),containsString("lucene:test-index(/oak:index/test-index)"));
+
+        List<String> result = executeQuery(query,SQL2);
+        assertEquals("Ordering doesn't match", asList("10 percent","10%","Hallo","hello","World!"), result);
+
+    }
+
+    @Test
+    public void testOrdering() throws Exception {
+        Tree luceneIndex = createIndex("upper", Collections.<String>emptySet());
+        Tree nonFunc = luceneIndex.addChild(FulltextIndexConstants.INDEX_RULES)
+                .addChild("nt:base")
+                .addChild(FulltextIndexConstants.PROP_NODE)
+                .addChild("foo");
+        nonFunc.setProperty(FulltextIndexConstants.PROP_ORDERED,true);
+        nonFunc.setProperty(FulltextIndexConstants.PROP_PROPERTY_INDEX,true);
+        nonFunc.setProperty("name", "foo");
+
+        Tree func = luceneIndex.getChild(FulltextIndexConstants.INDEX_RULES)
+                .getChild("nt:base")
+                .getChild(FulltextIndexConstants.PROP_NODE)
+                .addChild("fooUpper");
+        func.setProperty(FulltextIndexConstants.PROP_ORDERED,true);
+        func.setProperty(FulltextIndexConstants.PROP_FUNCTION, "fn:upper-case(@foo)");
+        func.setProperty(FulltextIndexConstants.PROP_PROPERTY_INDEX,true);
+
+        root.commit();
+
+        Tree test = root.getTree("/").addChild("test");
+        test.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+
+        List<String> paths = Lists.newArrayList();
+        for (int idx = 0; idx < 10; idx++) {
+            paths.add("/test/n" + idx);
+            if(idx%2 ==0) continue;
+            Tree a = test.addChild("n"+idx);
+            a.setProperty("jcr:primaryType", "nt:unstructured", Type.NAME);
+            a.setProperty("foo", "bar"+idx);
+
+        }
+        for(int idx =0 ; idx<10 ; idx++) {
```

Comment:
Formatting (spaces) are inconsistent, could you run the formatter please?

Reply:
Done